shared/teams/team/member/index.new.tsx

```diff
@@ -9,53 +9,120 @@ import * as ProfileGen from '../../../actions/profile-gen'
 import logger from '../../../logger'
 import {pluralize} from '../../../util/string'
 import {FloatingRolePicker} from '../../role-picker'
+import RoleButton from '../../role-button'
 import * as TeamsGen from '../../../actions/teams-gen'
 import {TeamDetailsSubscriber} from '../../subscriber'
+import {formatTimeForTeamMember} from '../../../util/timestamp'
 
 type Props = {
   teamID: Types.TeamID
   username: string
 }
 type OwnProps = Container.RouteProps<Props>
+type MetaPlusMembership = {
+  key: string
+  teamMeta: Types.TeamMeta
+  memberInfo: Types.MemberInfo
+}
+
+const getSubteamsInNotIn = (state: Container.TypedState, teamID: Types.TeamID, username: string) => {
+  const subteamsAll = [...Constants.getTeamDetails(state, teamID).subteams.values()]
+  let subteamsNotIn: Array<Types.TeamMeta> = []
+  let subteamsIn: Array<MetaPlusMembership> = []
+  subteamsAll.unshift(teamID)
+  for (const subteamID of subteamsAll) {
+    const subteamMeta = Constants.getTeamMeta(state, subteamID)
+    const subteamMembership = Constants.getTeamMembership(state, subteamID, username)
+
+    if (subteamMembership) {
+      subteamsIn.push({
+        key: `member:${username}:${subteamMeta.teamname}`,
+        memberInfo: subteamMembership,
+        teamMeta: subteamMeta,
+      })
+    } else {
+      subteamsNotIn.push(subteamMeta)
+    }
+  }
+  return {
+    subteamsIn,
+    subteamsNotIn,
+  }
+}
 
 const TeamMember = (props: OwnProps) => {
+  const dispatch = Container.useDispatch()
   const username = Container.getRouteProps(props, 'username', '')
   const teamID = Container.getRouteProps(props, 'teamID', Types.noTeamID)
+  const loading = Container.useAnyWaiting(Constants.loadSubteamMembershipsWaitingKey(teamID, username))
+
+  // Load up the memberships when the page is opened
+  React.useEffect(() => {
+    dispatch(TeamsGen.createGetMemberSubteamDetails({teamID, username}))
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
we're trying to avoid disbling this lint rule. You probably want to include `teamID` and `username` because if we switch to a different user we'll want to fetch the data again. Then throw in `dispatch` to make the linter happy.

Reply:
hmm i see, im 100% ok with an "implicit only run on mount" here

Reply:
@chrisnojima how strictly are we trying to follow this rule?